sdk/containerregistry/container-registry/review/container-registry.api.md

```diff
@@ -57,6 +57,22 @@ export interface ArtifactTagProperties {
     readonly repositoryName: string;
 }
 
+// @public
+export class ContainerRegistryBlobClient {
+    constructor(endpoint: string, repositoryName: string, credential: TokenCredential, options?: ContainerRegistryClientOptions);
+    deleteBlob(digest: string, options?: DeleteBlobOptions): Promise<void>;
+    deleteManifest(digest: string, options?: DeleteManifestOptions): Promise<void>;
+    downloadBlob(digest: string, options?: DownloadBlobOptions): Promise<DownloadBlobResult>;
+    downloadManifest(tagOrDigest: string, options?: DownloadManifestOptions): Promise<DownloadManifestResult>;
+    readonly endpoint: string;
+    readonly repositoryName: string;
+    uploadBlob(blobStreamFactory: () => NodeJS.ReadableStream): Promise<UploadBlobResult>;
+    uploadBlob(blobStream: NodeJS.ReadableStream): Promise<UploadBlobResult>;
+    uploadManifest(manifest: OciManifest, options?: UploadManifestOptions): Promise<UploadManifestResult>;
+    uploadManifest(manifestStream: NodeJS.ReadableStream, options?: UploadManifestOptions): Promise<UploadManifestResult>;
+    uploadManifest(manifestStreamFactory: () => NodeJS.ReadableStream, options?: UploadManifestOptions): Promise<UploadManifestResult>;
```

Comment:
Is it worth exposing these as overloads instead of something like 

```ts
uploadManifest(manifest: OciManifest | NodeJS.ReadableStream | () => NodeJS.ReadableStream`, options?: UploadManifestOptions): Promise<UploadManifestResult>
```

It doesn't seem like the doc comments for each overload are appreciably different and the parameters are identical except for the first one.

Reply:
I guess the value is that we can call out the resettable stream case (and it needing to return a fresh stream each time) more explicitly this way. No strong opinion here though.

Reply:
I updated it as per your suggestion.